### PR TITLE
fix: profile provider mutate

### PR DESCRIPTION
**Pull request by Jakub-Vacek**
## Description

This PR fixes problems with priority and defaults resolution when adding new profile provider

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have updated the documentation accordingly. If you are changing **code related to user secrets** you need to really make sure that [security documentation](SECURITY.md) is correct.
- [x] I have read the **CONTRIBUTION_GUIDE** document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/internal/superjson/mutate.ts [hunk 1]
```diff
@@ -257,32 +277,60 @@ export function addProfileProvider(
...
   if (typeof profileProvider === 'string') {
+    //Change
     defaults = payload.defaults;
-  } else if (profileProvider.defaults) {
-    if (!payload.defaults) {
-      defaults = targetedProfile.defaults;
-    } else {
+  } else {
+    if (profileProvider.defaults && payload.defaults) {
+      //Chanage
```
Comment (TheEdward162) [inline src/internal/superjson/mutate.ts:295]: Chanage lol

File: src/internal/superjson/mutate.ts [hunk 2]
```diff
@@ -220,6 +232,10 @@ export function addProfileProvider(
     targetedProfile.providers = {
       [providerName]: payload,
     };
+    targetedProfile.priority = resolvePriorityAddition(
+      targetedProfile.priority,
+      providerName
+    );
```
Comment (TheEdward162) [inline src/internal/superjson/mutate.ts:238]: Is this needed in this branch as we update the targetedProfile above?

**Review by lukas-valenta — APPROVED** [review]

File: src/internal/superjson/mutate.ts [hunk 3]
```diff
@@ -257,32 +277,60 @@ export function addProfileProvider(
...
   }
+  // if there no other keys and we changed defaults
+  if (
+    !('file' in payload) &&
+    !('mapVariant' in payload) &&
+    !('mapRevision' in payload) &&
+    defaults
+  ) {
+    targetedProfile.providers[providerName] = {
+      ...(typeof profileProvider === 'string' ? {} : profileProvider),
```
Comment (TheEdward162) [inline src/internal/superjson/mutate.ts:313]: Don't we need to preserve it as a `file` entry?

Comment (Jakub-Vacek) [reply src/internal/superjson/mutate.ts:313]: You mean like:
if typeof profileProvider === 'string' {
targetedProfile.providers[providerName] = {
file: profileProvider
defaults: payload.defaults
}
}

Comment (Jakub-Vacek) [reply src/internal/superjson/mutate.ts:313]: `    targetedProfile.providers[providerName] = {
      ...(typeof profileProvider === 'string' ? { file: profileProvider } : profileProvider),
      defaults,
    };`

Comment (TheEdward162) [reply src/internal/superjson/mutate.ts:313]: Yeah, otherwise I think we'd lose the path

**Review by TheEdward162 — APPROVED** [review]

**Pull request merged**